packages/core/src/test/awsService/cloudWatchLogs/commands/tailLogGroup.test.ts

```diff
@@ -0,0 +1,158 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as sinon from 'sinon'
+import * as vscode from 'vscode'
+
+import assert from 'assert'
+import { clearDocument, closeSession, tailLogGroup } from '../../../../awsService/cloudWatchLogs/commands/tailLogGroup'
+import { StartLiveTailResponseStream } from '@aws-sdk/client-cloudwatch-logs'
+import { LiveTailSessionRegistry } from '../../../../awsService/cloudWatchLogs/registry/liveTailSessionRegistry'
+import { LiveTailSession } from '../../../../awsService/cloudWatchLogs/registry/liveTailSession'
+import { asyncGenerator } from '../../../../shared/utilities/collectionUtils'
+import {
+    TailLogGroupWizard,
+    TailLogGroupWizardResponse,
+} from '../../../../awsService/cloudWatchLogs/wizard/tailLogGroupWizard'
+import { getTestWindow } from '../../../shared/vscode/window'
+
+describe('TailLogGroup', function () {
+    const testLogGroup = 'test-log-group'
+    const testRegion = 'test-region'
+    const testMessage = 'test-message'
+
+    let sandbox: sinon.SinonSandbox
+    let registry: LiveTailSessionRegistry
+    let startLiveTailSessionSpy: sinon.SinonSpy
+    let stopLiveTailSessionSpy: sinon.SinonSpy
+    let wizardSpy: sinon.SinonSpy
+
+    beforeEach(function () {
+        sandbox = sinon.createSandbox()
+        registry = new LiveTailSessionRegistry()
+    })
+
+    afterEach(function () {
+        sandbox.restore()
+    })
+
+    it('starts LiveTailSession and writes to document. Closes tab and asserts session gets closed.', async function () {
+        wizardSpy = sandbox.stub(TailLogGroupWizard.prototype, 'run').callsFake(async function () {
+            return getTestWizardResponse()
+        })
```

Comment:
we have a full wizard testing framework that allows the actual wizard code to run, and you can assert its state. please do not mock/spy things unless it's unavoidable (typically network/service calls).

Reply:
I have the TailLogGroup wizard under test in its own Test class https://github.com/aws/aws-toolkit-vscode/blob/feature/cwltail/packages/core/src/test/awsService/cloudWatchLogs/wizard/tailLogGroupWizard.test.ts
which is why i decided to Mock it here. This `tailLogGroup.test` was intended to test just the functionality of the command itself. 

In that case is it acceptable to leave this mocked? I can use that testing framework here also, was just trying to separate responsibilities of each testing class.

Reply:
> was just trying to separate responsibilities of each testing class.

Testing more things in one test can often be a benefit (less code, more coverage). Would it make sense to merge this test into the other? 

Or meanwhile, people tend to copy what exists. So the presence of mocks here encourages even more mocks...

Reply:
Sure, at the least I can copy over a basic form of the Wizard tester here to avoid mocking it. If I have a wizard tester in my test, how would you recommend supplying that to my Command? Doing local testing and the prompter still comes up when the test runs, waiting for input. 

Is it recommended to use something like dependency injection to provide a TailLogGroupWizard to the command in its constructor so the test can supply its Wizard to the command? 

I've added 
``` 
const wizard = new TailLogGroupWizard({
            groupName: testLogGroup,
            regionName: testRegion,
        })
        const tester = await createWizardTester(wizard)
        tester.regionLogGroupSubmenuResponse.assertDoesNotShow()
        tester.logStreamFilter.assertShowFirst()
        tester.filterPattern.assertShowSecond()
```
to my test before executing the command `await tailLogGroup(registry, {
            groupName: testLogGroup,
            regionName: testRegion,
        })`, but its hanging waiting for input on the Wizard because the command is creating a new Wizard and calling prompt on it.

Reply:
> but its hanging waiting for input on the Wizard because the command is creating a new Wizard and calling prompt on it

the wizard tester has functions to choose items from it. look at examples in other tests